File: 3649/3649.cpp
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

void twoPointer(vector<int> v, int len)
{
  int l = 0, r = v.size() - 1;
  while (l < r)
  {
    int sum = v[l] + v[r];
    if (sum == len)
    {
      cout << "yes " << v[l] << ' ' << v[r] << '\n';
      return;
    }
    if (sum < len)
    {
      l += 1;
    }
    else if (sum > len)
    {
      r -= 1;
    }
  }
  cout << "danger\n";
}

int main()
{
  ios::sync_with_stdio(false);
  cin.tie(0);
  cout.tie(0);

  int NANO = 10000000;
  int len;

  while (cin >> len)
  {
    int size;
    cin >> size;

    vector<int> v;
    for (int i = 0; i < size; i++)
    {
      int x;
      cin >> x;
      v.push_back(x);
    }
    sort(v.begin(), v.end());
    twoPointer(v, len * NANO);
  }
}

File: 3649/3649.js
//정렬 후 투포인터
const fs = require('fs');
const input = fs.readFileSync('input.txt').toString().split('\n');

let idx = 0;
const NANO = 10_000_000;
while (idx < input.length) {
  const len = +input[idx++] * NANO;
  const size = +input[idx++];

  const q = [];
  for (let i = 0; i < size; i++) {
    q.push(+input[idx++]);
  }
  q.sort((a, b) => a - b);
  twoPointer(q, len);
}

function twoPointer(q, len) {
  let [l, r] = [0, q.length - 1];
  while (l < r) {
    const sum = q[l] + q[r];
    if (sum === len) {
      console.log('yes', q[l], q[r]);
      return;
    }
    if (sum < len) {
      l += 1;
    } else if (sum > len) {
      r -= 1;
    }
  }
  console.log('danger');
}
